### ISSUE TITLE
Route53 support

### ISSUE THREAD

**Issue by ytsarev** [AbsaOSS/k8gb#162]
To enable wider k8gb adoption we need to add more common edgeDNS provider like AWS Route53

* Extend k8gb with ability to automatically configure DNS zone delegation with Route53. Similar to functionality implemented with Infoblox API right now
* Possibly use additional instance of external-dns for configuration assuming no special api calls apart from NS and A creation required.

**Comment by adrienjt** [comment]
I like the external-dns option. It would be compatible with many DNS providers. What would it take to implement it? Are there currently "special API calls apart from NS and A creation"? If I'm not mistaken, looking at the Git log, you used to go through external-dns to integrate with Infoblox, but switched to using the Infoblox API directly. Why?

**Comment by ytsarev** [comment]
@adrienjt you are totally right, the first attempt was to utilize external-dns for infoblox as well. We had to switch from this approach in case of Infoblox because it has special ZoneDelegated api call which creates NS and A records for name servers in one shot behind the scenes. Because of the fact that we could not create zone delegation with direct creation of DNS records( which would fit external-dns perfectly) so we had to replace it with direct api calls with infoblox-go-client.

I am evaluating route53 and overall external-dns integration for edgeDNS right now - unfortunately zone delegation config is inconsistent from provider to provider
